managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -622,9 +622,11 @@ public void asyncReadEntries(int numberOfEntriesToRead, long maxSizeBytes, ReadE
 
         int numOfEntriesToRead = applyMaxSizeCap(numberOfEntriesToRead, maxSizeBytes);
 
-        PENDING_READ_OPS_UPDATER.incrementAndGet(this);
         OpReadEntry op = OpReadEntry.create(this, readPosition, numOfEntriesToRead, callback, ctx, maxPosition);
-        ledger.asyncReadEntries(op);
+        if (op.readPosition != null) {
```

Comment:
op is not recycle if op.readPosition == null

By the way, could you add some unit test?

Reply:
op recycled when the method `OpReadEntry->readEntriesFailed` called.

At present, op init failed only if `op.readPosition = cursor.ledger.startReadOperationOnLedger(readPositionRef, op);` called.

I do not know how to test this case ...